apps/cowswap-frontend/src/modules/hooksStore/pure/AddCustomHookForm/CustomDappLoader/index.tsx

```diff
@@ -25,27 +42,53 @@ export function ExternalDappLoader({
 }: ExternalDappLoaderProps) {
   const { chainId } = useWalletInfo()
 
-  useEffect(() => {
-    let isRequestRelevant = true
+  const fetchManifest = useCallback(
+    async (url: string) => {
+      if (!url) return
+
+      setLoading(true)
+
+      try {
+        const validation = validateHookDappUrl(url)
+        if (!validation.isValid) {
+          setManifestError(validation.error)
+          setDappInfo(null)
+          setLoading(false)
+          return
+        }
+
+        const trimmedUrl = url.trim()
+        const manifestUrl = `${trimmedUrl}${trimmedUrl.endsWith('/') ? '' : '/'}manifest.json`
 
-    setLoading(true)
+        const response = await fetchWithTimeout(manifestUrl, { timeout: TIMEOUT })
+        if (!response.ok) {
+          setManifestError('Failed to fetch manifest. Please verify the URL and try again.')
+          setDappInfo(null)
+          setLoading(false)
+          return
+        }
+
+        const data = await response.json()
 
-    fetch(`${input}/manifest.json`)
-      .then((res) => res.json())
-      .then((data) => {
-        if (!isRequestRelevant) return
+        if (!data.cow_hook_dapp) {
+          setManifestError('Invalid manifest format: missing cow_hook_dapp property')
+          setDappInfo(null)
+          setLoading(false)
```

Comment:
nit: This pattern is repeated 3 times. Maybe consider to do a refactor?

```js
const setError = (message: string) =>  {
  setManifestError(message)
  setDappInfo(null)
  setLoading(false)
}
```

Reply:
Refactored this now